common/types/int.go

```diff
@@ -75,14 +86,32 @@ func (i Int) Compare(other ref.Val) ref.Val {
 func (i Int) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
 	switch typeDesc.Kind() {
 	case reflect.Int32:
+		// Enums are also mapped as int32 derivations.
 		return reflect.ValueOf(i).Convert(typeDesc).Interface(), nil
 	case reflect.Int64:
 		return int64(i), nil
 	case reflect.Ptr:
-		if typeDesc == jsonValueType {
+		switch typeDesc {
+		case anyValueType:
+			return ptypes.MarshalAny(&wrapperspb.Int64Value{Value: int64(i)})
+		case int32WrapperType:
```

Comment:
Add a note that we can't export to EnumValue since we don't know which enum we'd be talking about.